espnet2/train/trainer.py

```diff
@@ -359,12 +364,15 @@ def run(
                     reporter.wandb_log()
 
                 # 4. Save/Update the checkpoint
-                if use_lora and save_lora_only:
-                    # Only the LoRA realted params are saved, not the whole model
-                    model_state_dict = lora.lora_state_dict(model)
-                else:
-                    # Save all params of the model
-                    model_state_dict = model.state_dict()
+                # TODO: This kind of design may not facilitate the integration of other kind of the adapter
+                # Maybe we can only save those with requires_grad = True ?
+                # If we use lora.lora_state_dict, we will not save the downstream model in SSL settings
```

Comment:
I do not have a clear question to this one. But I feel only save models with requires_grad = True is a bit risky.

Ping @sw005320 @wanchichen @simpleoier @pyf98 for some discussion here.

Reply:
I think so. One case is that it can fail if the pretrained SSL models are updated.

Reply:
For that purpose, I would suggest we simply follow the setting with the current design.

Reply:
The reason why I choose to save all parameters with `requires_grad = True' is that in adapter setting, we usually need to save the downstream model or other tunable parameters together (Ex, the weighted_sum). Therefore, we may not able to follow what loralib did, which only save lora_layers. 
I understand your concern about the current design. However, for the case SSL models are updated, I think it would be better to save the SSL weights (or I can not see why we need to update it without saving it). If we have to follow what loralib does by offering an options which saves the adapter parameter only, then I   would like to add another option for saving parameters with `requires_grad = True'. In this case, we can fulfill the requirement but also make sure adapter and lora can work in s3prl settings.
Do you think it's a good idea?

Reply:
I made some revisions here.
Please check [d7f0b39](https://github.com/espnet/espnet/pull/5609/commits/d7f0b399dd47ccb7dd2714b0b584fad8b5a3c2be)

@ftshijt @simpleoier